client/CPlayerInterface.cpp

```diff
@@ -1685,16 +1686,39 @@ void CPlayerInterface::showShipyardDialogOrProblemPopup(const IShipyard *obj)
 
 void CPlayerInterface::requestReturningToMainMenu(bool won)
 {
+	HighScoreParameter param;
+	param.difficulty = cb->getStartInfo()->difficulty;
+	param.day = cb->getDate();
+	param.townAmount = cb->howManyTowns();
+	param.usedCheat = cb->getPlayerState(*cb->getPlayerID())->enteredWinningCheatCode;

```

Comment:
This only checks for one cheat code. Check PlayerMessageProcessor::cheaters.
(only available on server side)

Reply:
added a generic cheater flag in playerstate.